src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -146,11 +147,19 @@ private DashboardCommands parseDashboardCommandFromInput(String input) {
      * @param input full user input string
      * @return module code string
      */
-    private String getModuleCode(String input) {
-        // TODO  - error handling
+    private String getModuleCode(String input) throws CommandException {
         String[] words = input.split(" ");
+        if (words.length < 2) {
+            throw new CommandException("Module not specified.");
+        }
+
+        String moduleCode = words[1].toUpperCase();
+
+        if (!isValidModuleCode(moduleCode)) {
```

Comment:
In my view, the isValidModuleCode method should be in the Module class as a class-level method since it could be a general method that other components might need to use. i.e. loading of module lists.

Reply:
We could move it there, but it has to be public maybe so everyone can use it?